packages/scripts/src/split-file/get-code-sections.js

```diff
@@ -0,0 +1,69 @@
+import { functionTypeRegex } from './utils.js';
+
+/**
+ * @param {{block: string, typeIndex: number | null}[]} blocks
+ * @param {string[]} typeStrings
+ * @returns {string[]}
+ */
+const getUnassignedTypes = (blocks, typeStrings) => {
+	const assignedTypesIndices = blocks.reduce((/** @type {number[]} */ acc, { typeIndex }) => {
+		if (!typeIndex) return acc;
+		return [...acc, typeIndex];
+	}, []);
+	return typeStrings.filter((_, ii) => !assignedTypesIndices.includes(ii));
+};
+
+/**
+ * @param {{block: string, typeIndex: number | null}[]} blocks
+ * @param {string[]} typeStrings
+ * @returns {{block: string, type: string | null}[]}
+ */
+const writeTypesToBlocks = (blocks, typeStrings) =>
+	blocks.map(({ block, typeIndex }) => ({
+		block,
+		type: typeIndex !== null ? typeStrings[typeIndex] : null
+	}));
+/**
+ * @param {import("./cursor.js").Cursor} cursor
+ */
+export const getCodeSections = (cursor) => {
```

Comment:
Puts the various types of blocks into separate arrays for the file writer to use later